traceon/geometry.py

```diff
@@ -673,6 +670,126 @@ def aperture(height, radius, extent, z=0.):
+            theta = l/radius
+            return center + np.cos(theta) * center_to_start + np.sin(theta)*np.cross(plane_normal, center_to_start)
+        
+        return Path(f, radius*angle)
+    
+    def arc_to_polar(self, radius, angle, plane_normal=[0, 1, 0]):
+        """Extend the current path by a smooth arc using polar coordinates.
+        The arc is defined by a specified radius and angle and rotates counterclockwise
+         around around the normal that defines the arcing plane.
+
+        Parameters
+        ---------------------------
+        radius : float
+            The radius of the arc
+        angle : float
+            The angle subtended by the arc (in radians)
+        plane_normal : (3,) float
+            The normal vector of the plane containing the arc
+
+        Returns
+        ----------------------------
+        Path"""
+        plane_normal = np.array(plane_normal, dtype=float)
+        start_point = self.endpoint()
+        tangent = self.tangent(self.path_length)
+
+        dot_product = np.dot(tangent, plane_normal)
+        if not np.isclose(dot_product, 0,atol=1e-7):
+
+            corrected_normal = plane_normal - (dot_product / np.linalg.norm(tangent)**2) * tangent
+            raise AssertionError(
+                f"The provided plane normal {plane_normal} is not orthogonal to the tangent {tangent}  \n"
+                f"of the path at the endpoint so no smooth arc can be made. The closest valid normal is "
+                f"{np.round(corrected_normal, 10)}.")
+        
+        plane_normal /= np.linalg.norm(plane_normal)
+
+        return self >> Path.arc_polar(radius, angle, start_point, plane_normal, tangent)
+
+    def reversed(self):
+        """Generate a reversed version of the current path.
+        The reversed path is created by inverting the traversal direction,
+        such that the start becomes the end and vice versa.
+
+        Returns
+        ----------------------------
+        Path"""
+        return Path(lambda t: self(self.path_length-t), self.path_length, self.breakpoints, self.name)
+    
+    def tangent(self, t, num_splines=1000):
+        """Calculate the tangent vector at a specific point on the path 
+        using cubic spline interpolation.
+
+        Parameters
+        ----------------------------
+        t : float
+            The point on the path at which to calculate the tangent
+        num_splines : int
+            The number of samples used for cubic spline interpolation
+
+        Returns
+        ----------------------------
+        (3,) np.ndarray of float"""
+
+        samples = np.linspace(0, self.path_length, num_splines) 
+        
+        # Include t to slightly improve accuracy
+        if not t in samples:
+            samples = np.append(samples, t)
+            samples = np.sort(samples)
+
+        splines = CubicSpline(samples, [self(s) for s in samples])
+        tangent = splines.derivative()(t)
+        
+        return tangent
```

Comment:
Consider the below implementation, which is simpler and doens't need so many samples
```Python
samples = np.linspace(t - self.path_length*1e-3, t + self.path_length*1e-3, 7) # Odd number to include t
samples_on_path = [s for s in samples if 0 <= s <= self.path_length]
assert len(samples_on_path), "Please supply a point that lies on the path"
return CubicSpline(samples_on_path, [self(s) for s in samples_on_path])(t, nu=1)
```

Reply:
I have thought about using a local spline as well. My concern was that it might lead to small inconsistencies in the tangent, since the spline is different for each t, but I guess this is the case anyway since we include t. What do you think?

Reply:
I don't see what inconsistencies would arise? Can you give an example?

Reply:
Inconsistencies is maybe not the right word. My reasoning was that since two points  which are epsilon apart, create two different sets of splines, the tangent vector is computed at a ever so slightly different spline, which might lead to the computed tangent along the entire path to be less smooth then when it is computed from global splines. But to be honest I think the effect is neglible. On the other hand, the computation of 1000 splines also was not very expensive in my experience. However, I will change it to your suggested implementation

Reply:
Sounds good, I think the 'non smoothness' will be negligible